Source/WTF/wtf/StackTrace.cpp

```diff
@@ -45,6 +45,55 @@ void WTFGetBacktrace(void** stack, int* size)
 
 namespace WTF {
 
+#if USE(LIBBACKTRACE)
+static struct backtrace_state* backtraceState()
+{
+    static struct backtrace_state* backtraceState = backtrace_create_state(nullptr, 1, nullptr, nullptr);
+    return backtraceState;
+}
+
+static void backtraceSyminfoCallback(void* data, uintptr_t, const char* symname, uintptr_t, uintptr_t)
+{
+    const char** symbol = static_cast<const char**>(data);
+    *symbol = symname;
+}
+
+static int backtraceFullCallback(void* data, uintptr_t, const char*, int, const char* function)
+{
+    const char** symbol = static_cast<const char**>(data);
+    *symbol = function;
+    return 0;
+}
+
+const char** symbolize(void* const* addresses, int size)
+{
+    struct backtrace_state* state = backtraceState();
+    if (!state)
+        return nullptr;
+
+    const char** symbols = (const char**)malloc(sizeof(const char*) * size);
+
+    for (int i = 0; i < size; ++i) {
+        uintptr_t pc = reinterpret_cast<uintptr_t>(addresses[i]);
+        const char* symbol;
+
+        backtrace_pcinfo(state, pc, backtraceFullCallback, nullptr, &symbol);
+        if (!symbol)
+            backtrace_syminfo(backtraceState(), pc, backtraceSyminfoCallback, nullptr, &symbol);
+
+        if (symbol) {
+            char* demangled = abi::__cxa_demangle(symbol, nullptr, nullptr, nullptr);
+            if (demangled)
+                symbols[i] = demangled;
+            else
+                symbols[i] = symbol;
+        } else
+            symbols[i] = "???";
```

Comment:
Since the result of `abi::__cxa_demangle` needs to be freed, you'll need to ensure these strings are heap-allocated too. Normally we would use `g_strdup` for this, but since this code doesn't use GLib you'll want `strdup` from string.h:

```suggestion
                symbols[i] = strdup(symbol);
        } else
            symbols[i] = strdup("???");
```

Reply:
You're right. Done.